src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -152,14 +138,20 @@ public function validate($entity, Constraint $constraint)
             reset($result);
         }
 
-        /* If no entity matched the query criteria or a single entity matched,
-         * which is the same as the entity being validated, the criteria is
-         * unique.
-         */
-        if (0 === count($result) || (1 === count($result) && $entity === ($result instanceof \Iterator ? $result->current() : current($result)))) {
+        if (0 === count($result)) {
             return;
         }
 
+        if (1 === count($result)) {
+            if ($constraint->forUpdate) {
```

Comment:
what about checking the corresponding source fields against the entity fields instead?

Reply:
I don't see how this could work?

Reply:
for each source field get the entity field value, compare entity field values against source field values. But im not sure thats waterproof; nor is `forUpdate`.

Reply:
If I want to update an entity without changing a unique constrained field then this wouldn't work, that's why I let the decision to the developer.